Snakefile

```diff
@@ -0,0 +1,106 @@
+N_FILES_MAX_PER_SAMPLE = -1
+download_sleep = 0
+url_prefix = "root://eospublic.cern.ch//eos/opendata"
+#In order to run analysis from Nebraska use this prefix
+#url_prefix = "https://xrootd-local.unl.edu:1094//" 
+import glob
+import json
+import os
+def extract_samples_from_json(json_file):
+    output_files = []
+    
+    with open(json_file, "r") as fd:
+        data = json.load(fd)
+
+        for sample, conditions in data.items():
+            for condition, details in conditions.items():
+                sample_name = f"{sample}__{condition}"
+                output_files.append(sample_name)
+                with open(f"sample_{sample_name}_paths.txt", "w") as path_file:
+                    paths = [file_info["path"].replace("https://xrootd-local.unl.edu:1094//store/user/AGC/nanoAOD",
+                                                       "root://eospublic.cern.ch//eos/opendata/cms/upload/agc/1.0.0/") for file_info in details["files"]]
+                    path_file.write("\n".join(paths))
+
+    return output_files
+    
+def get_file_paths(wildcards, max=N_FILES_MAX_PER_SAMPLE):
+    "Return list of at most MAX file paths for the given SAMPLE."
+    import json
+    import os
+    filepaths = []
+    fd = open(f"sample_{wildcards.sample}__{wildcards.condition}_paths.txt")
+    filepaths = fd.read().splitlines()
+    fd.close()
+    return [f"histograms/histograms_{wildcards.sample}__{wildcards.condition}__"+filepath[38:] for filepath in filepaths][:max] 
+
+samples = extract_samples_from_json("nanoaod_inputs.json")
+
+def get_items(json_file):
```

Comment:
Note that I'm getting a bunch of warnings as follows when submitting the workflow for execution:

```
File path histograms/histograms_wjets__nominal__/cms/upload/agc/1.0.0//WJetsToLNu_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/cmsopendata2015_wjets_20548_PU25nsData2015v1_76X_mcRun2_asymptotic_v12_ext4-v1_80000_0024.root contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
[CRITICAL] File path histograms/histograms_wjets__nominal__/cms/upload/agc/1.0.0//WJetsToLNu_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/cmsopendata2015_wjets_20548_PU25nsData2015v1_76X_mcRun2_asymptotic_v12_ext4-v1_80000_0024.root contains double '/'. This is likely unintended. It can also lead to inconsistent results of the file-matching approach used by Snakemake.
```